engine/access/subscription/block_tracker.go

```diff
@@ -0,0 +1,128 @@
+package subscription
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/counters"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/state/protocol"
+
+	"github.com/onflow/flow-go/storage"
+)
+
+// GetHighestHeight is a function type for getting the highest height.
+// Block status could be only BlockStatusSealed or BlockStatusFinalized.
+type GetHighestHeight func(flow.BlockStatus) (uint64, error)
+
+// BlockTracker is an interface for tracking blocks and handling block-related operations.
+type BlockTracker interface {
+	BaseTracker
+	// GetHighestHeight returns the highest height based on the specified block status which could be only BlockStatusSealed
+	// or BlockStatusFinalized.
+	// No errors are expected during normal operation.
+	GetHighestHeight(flow.BlockStatus) (uint64, error)
+	// ProcessOnFinalizedBlock drives the subscription logic when a block is finalized.
+	// The input to this callback is treated as trusted. This method should be executed on
+	// `OnFinalizedBlock` notifications from the node-internal consensus instance.
+	ProcessOnFinalizedBlock() error
```

Comment:
```suggestion
	// No errors are expected during normal operations.
	ProcessOnFinalizedBlock() error
```

I suspect no errors are expected?